hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/TypedTable.java

```diff
@@ -71,6 +96,27 @@ public boolean isEmpty() throws IOException {
 
   @Override
   public VALUE get(KEY key) throws IOException {
+    // Here the metadata lock will guarantee that cache is not updated for same
+    // key during get key.
+    if (cache != null) {
+      CacheValue<VALUE> cacheValue = cache.get(new CacheKey<>(key));
+      if (cacheValue == null) {
+        return getFromTable(key);
+      } else {
+        // Doing this because, if the Cache Value Last operation is deleted
+        // means it will eventually removed from DB. So, we should return null.
+        if (cacheValue.getLastOperation() != CacheValue.OperationType.DELETED) {
+          return cacheValue.getValue();
+        } else {
+          return null;
+        }
+      }
+    } else {
+      return getFromTable(key);
```

Comment:
Not sure if you need this get again ?

Reply:
For tables where the cache is disabled, we need to do as before just read from DB and return data.

Reply:
Understood the comment, updated the code to remove getTable in multiple places.